File: atcoder/abc171C_one_quadrillion_and_one_dalmatians.py
# Implementation by unihernandez22 || Idea by Vicfred
# https://atcoder.jp/contests/abc171/tasks/abc171_c
# math, greedy

n = int(input())

digits = []

while n > 0:
    n -= 1
    digits.append(chr(n%26+97))
    n //= 26

print(''.join(reversed(digits)))

File: atcoder/abc171C_one_quadrillion_and_one_dalmatians.nim
# Vicfred
# https://atcoder.jp/contests/abc171/tasks/abc171_c
# implementation, math
import strutils, algorithm

var n = stdin.readLine.parseInt

var ans = ""

while n > 0:
  n -= 1
  ans &= char(ord('a') + n mod 26)
  n = n div 26

ans.reverse
echo ans

